### PR TITLE
Fix compile warnings breaking builds with -Werror.

### PR DESCRIPTION
**Pull request by EmilyBjoerk**
Ran into some problems pulling in hueplusplus into my project that uses strict flags for warnings.

Fixes some sign mismatches and implicit fall through and missing initializer.

### CODE REVIEW

**Comment by enwi** [comment]
Hi @EmilyBjoerk and thanks for the work. Have you checked if a modification of the included [json library](https://github.com/open-source-parsers/jsoncpp) is really necessary? The version we are using here is pretty old now and I saw a [PR on that topic](https://github.com/open-source-parsers/jsoncpp/pull/697) -> will this fix the issue as well? Then we can just update the library

File: hueplusplus/LinHttpHandler.cpp [hunk 1]
```diff
@@ -98,18 +98,18 @@ std::string LinHttpHandler::send(const std::string &msg, const std::string &adr,
...
+    else {
       sent += bytes;
     }
   } while (sent < total);
 
   // receive the response
   std::string response;
   total = sizeof(response) - 1;
-  int received = 0;
+  decltype(total) received = 0;
```
Comment (enwi) [inline hueplusplus/LinHttpHandler.cpp]: Why not use size_t here?
```suggestion
  size_t received = 0;
```

Comment (EmilyBjoerk) [reply hueplusplus/LinHttpHandler.cpp]: It was 4:30 AM after an all-nighter, I was in an unfamiliar codebase and took the easiest solution to fixing the compile warning blocking my project from deploying before I could go to bed. :)

I'll fix in a bit.

**Comment by EmilyBjoerk** [comment]
> Hi @EmilyBjoerk and thanks for the work.
> 
> Have you checked if a modification of the included [json library](https://github.com/open-source-parsers/jsoncpp) is really necessary?
> 
> The version we are using here is pretty old now and I saw a [PR on that topic](https://github.com/open-source-parsers/jsoncpp/pull/697) -> will this fix the issue as well? Then we can just update the library

I think pulling a newer version from upstream might fix it, we can try that first.

**Comment by EmilyBjoerk** [comment]
Please let me know when you have updated the json library from upstream, their file structure seemed significantly different from what I saw in your repo. I'm not sure if they compile the source into one file or if your version is prior some refactoring so I'm not confident doing the update in this PR.

**Comment by enwi** [comment]
I switched to another JSON library nlohmann-json, which I wanted to do for some time now

**Comment by EmilyBjoerk** [comment]
I'm unfamiliar with your git workflow, looking at the history I would have expected to see back merges from master to development but it seems you rebase/cherrypick instead? I'm not quite sure how you want me to do this.

The changes I've made are basically trivial and I wonder if it isn't easier for you to just replicate them with your familiar workflows at this point? I've forked and fixed what I needed on my fork and now completed/deployed my project already so I'm not that concerned one way or another about the contribution "credit".

**Comment by enwi** [comment]
@EmilyBjoerk Nö worries, changes are pushed and pullrequests are merges with the development branch first, then once the changes have been checked and they're stable they get merged into the master branch technically creating a.new version of the library. Maybe it's time to create a contribution guide.

**Comment by enwi** [comment]
@EmilyBjoerk I've tested your code with the changes made on the development branch and it all works fine.

**Pull request closed without merging**